desktop-app/app/components/ScreenshotManager/Capture.js

```diff
@@ -0,0 +1,268 @@
+  if (!sortedWebViews.length) return;
+
+  // remove event mirroring between web views
+  pubsub.publish(TOGGLE_EVENT_MIRRORING_ALL_DEVICES, [{status: false}]);
+  // set loader in each webview
+  pubsub.publish(SCREENSHOT_IN_PROGRESS, [{isInProgress: true}]);
+  // store image objects
+  const images = [];
+  // device names
+  const deviceNames = [];
+
+  // get web address of Web views
+  const address = sortedWebViews[0].getURL();
+
+  const fsUtils = new FileSystemUtils(path, defaultPath, now, address);
+
+  for (let i = 0; i < sortedWebViews.length; i += 1) {
+    const toastId = showNotification(
+      'start Capturing',
+      false,
+      true,
+      false,
+      null
+    );
+    try {
+      const currentView = sortedWebViews[i];
+      currentView.scrollIntoView();
+      showNotification(
+        `Capturing ${devices[i].name}`,
+        false,
+        true,
+        false,
+        'info',
+        toastId
+      );
+      const webUtils = new WebViewUtils(currentView);
+      const image = await webUtils.captureWithRetry();
+      images.push(image);
+
+      const jpg = convertNativeImageToJPEG(image);
+      const deviceName = deviceChecks[i].name;
+      deviceNames.push(deviceName);
+      await fsUtils.writeImageToFile(jpg, deviceName, 'jpg');
+      showNotification(
+        `captured and saved: ${devices[i].name}`,
+        false,
+        false,
+        true,
+        'success',
+        toastId
+      );
+    } catch (error) {
+      showNotification(
+        `${error.message}: ${devices[i].name}`,
+        false,
+        false,
+        false,
+        'error',
+        toastId
+      );
+    }
+    toast.dismiss();
+  }
+
+  // remove event mirroring between web views
+  pubsub.publish(TOGGLE_EVENT_MIRRORING_ALL_DEVICES, [{status: true}]);
+  // remove loader
+  pubsub.publish(SCREENSHOT_IN_PROGRESS, [{isInProgress: false}]);
+
+  // account for browser to settle
+  await _delay(100);
+
+  if (isMergeImages) {
+    const toastId = showNotification(
+      'start Capturing',
+      false,
+      true,
+      false,
+      null
+    );
+    showNotification(`merging images`, false, true, false, 'info', toastId);
+    const mergedImage = await mergeImages(images, deviceNames);
+    showNotification(
+      `Converting data to binary`,
+      false,
+      true,
+      false,
+      'info',
+      toastId
+    );
+    const [img, ext] = await convertURIImageToBuffer(mergedImage);
+    showNotification(`saving`, false, true, false, 'info', toastId);
+    await fsUtils.writeImageToFile(img, 'all', ext);
+    showNotification(
+      `saved merged images`,
+      2000,
+      false,
+      true,
+      'success',
+      toastId
+    );
+  }
+
+  // open directory
+  await fsUtils.openCurrentDir();
+}
+
+// noinspection JSUndefinedPropertyAssignment
+export async function mergeImages(
+  images: Array<Electron.NativeImage>,
+  deviceNames: Array<string>
+) {
+  const canvas = document.createElement('canvas');
```

Comment:
@durgakiran Is there a better way to do the merging instead of using a `canvas` in terms of performance? 
Just to make sure the screenshot functionality finishes as quickly as possible.

Reply:
@manojVivek I think, we should really explore here. So far I did not find any, but I believe there can be better methods. maybe we should explore a bit of web assembly.

Reply:
I am exploring `webGL`, seems promising. I will update you if I find a real good way.

Reply:
I have optimized the process a bit, the merging and saving is fast now.